binfmt/binfmt_execmodule.c

```diff
@@ -285,8 +285,14 @@ int exec_module(FAR struct binary_s *binp,
 
 #if defined(CONFIG_ARCH_ADDRENV) || defined(CONFIG_ARCH_VMA_MAPPING)
 errout_with_tcbinit:
-  tcb->cmn.stack_alloc_ptr = NULL;
-  nxsched_release_tcb(&tcb->cmn, TCB_FLAG_TTYPE_TASK);
+#ifndef CONFIG_BUILD_KERNEL
+  if (binp->stackaddr != NULL)
+    {
+      tcb->cmn.stack_alloc_ptr = NULL;
+    }
+#endif
```

Comment:
Can you please explain why this was broken.

The process will be destroyed anyway -> the entire address environment is destroyed -> the stack gets freed when this happens. What am I missing ?

Reply:
This doesn't break anything, I agree with you, when the address environment is destroyed, we don't need to care about the unreleased memory, but according to the code logic, it is more reasonable to do the corresponding release in nxtask_uninit()

Reply:
Ok, thanks for the explanation.

One note: when freeing stack_alloc_ptr you must be sure the correct address environment is instantiated, which in this case it is, so the change is fine!